File: CS/GridMVVMBandsSample/EmployeesViewModel.cs
using DevExpress.Mvvm;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace GridMVVMBandsSample {
    public class ViewModel : ViewModelBase {
        public ViewModel() {
            Source = EmployeesDataModel.GetEmployees();
            Bands = new ObservableCollection<Band>() {
                new Band(
                    "Personal Info",
                    new ObservableCollection<Column>(),
                    new ObservableCollection<Band>() {
                        new Band (
                            "Name",
                            new ObservableCollection<Column>() {
                                new Column(nameof(Employee.FirstName)),
                                new Column(nameof(Employee.LastName))
                            },
                            new ObservableCollection<Band>()
                        ),
                        new Band (
                            "Birthday",
                            new ObservableCollection<Column>() {
                                new Column(nameof(Employee.BirthDate))
                            },
                            new ObservableCollection<Band>()
                        )
                    }
                ),
                new Band(
                    "Location",
                    new ObservableCollection<Column>() {
                        new Column(nameof(Employee.Country)),
                        new Column(nameof(Employee.City))
                    },
                    new ObservableCollection<Band>()
                ),
                new Band(
                    "Position",
                    new ObservableCollection<Column>() {
                        new Column(nameof(Employee.JobTitle))
                    },
                    new ObservableCollection<Band>()
                )
            };
        }
        public List<Employee> Source { get; }
        public ObservableCollection<Band> Bands { get; }
    }
    public class Column : BindableBase {
        public Column(string fieldname) {
            FieldName = fieldname;
        }
        public string FieldName { get; }
    }
    public class Band : BindableBase {
        public Band(string header, ObservableCollection<Column> childcolumns, ObservableCollection<Band> childbands) {
            Header = header;
            ChildColumns = childcolumns;
            ChildBands = childbands;
        }
        public string Header { get; }
        public ObservableCollection<Column> ChildColumns { get; }
        public ObservableCollection<Band> ChildBands { get; }
    }
}

File: VB/GridMVVMBandsSample/EmployeesViewModel.vb
Imports DevExpress.Mvvm
Imports System.Collections.Generic
Imports System.Collections.ObjectModel

Namespace GridMVVMBandsSample
	Public Class ViewModel
		Inherits ViewModelBase

		Public Sub New()
			Source = EmployeesDataModel.GetEmployees()
			Bands = New ObservableCollection(Of Band)() _
				From {
					New Band("Personal Info", New ObservableCollection(Of Column)(), New ObservableCollection(Of Band)() _
						From {
							New Band("Name", New ObservableCollection(Of Column)() From {
								New Column(NameOf(Employee.FirstName)),
								New Column(NameOf(Employee.LastName))
							}, New ObservableCollection(Of Band)()),
							New Band("Birthday", New ObservableCollection(Of Column)() From {New Column(NameOf(Employee.BirthDate))}, New ObservableCollection(Of Band)())
						}),
					New Band("Location", New ObservableCollection(Of Column)() From {
						New Column(NameOf(Employee.Country)),
						New Column(NameOf(Employee.City))
					}, New ObservableCollection(Of Band)()),
					New Band("Position", New ObservableCollection(Of Column)() From {New Column(NameOf(Employee.JobTitle))}, New ObservableCollection(Of Band)())
				}
		End Sub
		Public ReadOnly Property Source() As List(Of Employee)
		Public ReadOnly Property Bands() As ObservableCollection(Of Band)
	End Class
	Public Class Column
		Inherits BindableBase

		Public Sub New(ByVal fieldname As String)
			Me.FieldName = fieldname
		End Sub
		Public ReadOnly Property FieldName() As String
	End Class
	Public Class Band
		Inherits BindableBase

		Public Sub New(ByVal header As String, ByVal childcolumns As ObservableCollection(Of Column), ByVal childbands As ObservableCollection(Of Band))
			Me.Header = header
			Me.ChildColumns = childcolumns
			Me.ChildBands = childbands
		End Sub
		Public ReadOnly Property Header() As String
		Public ReadOnly Property ChildColumns() As ObservableCollection(Of Column)
		Public ReadOnly Property ChildBands() As ObservableCollection(Of Band)
	End Class
End Namespace
